src/libclipper/src/redis.cpp

```diff
@@ -145,10 +145,8 @@ std::vector<VersionedModelId> str_to_models(const std::string& model_str) {
         start +
         model_str.substr(start, end - start).find(ITEM_PART_CONCATENATOR);
```

Comment:
This method will break if `ITEM_PART_CONCATENATOR` or `ITEM_DELIMITER` are characters in the model name or version. We should check to make sure that when models are registered, the name and version don't contain these characters. We should have already been doing this but looks like it got overlooked.